### PR TITLE
Transform feedback interleaved

### PR DESCRIPTION
**Pull request by shrekshao**
Quite many changes in pipeline: use a new create program function

### CODE REVIEW

File: samples/transform_feedback_interleaved.html [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+        var programs = (function(gl, vertexShaderSourceTransform, fragmentShaderSourceTransform, vertexShaderSourceFeedback, fragmentShaderSourceFeedback) {
+            function createShader(gl, source, type) {
+                var shader = gl.createShader(type);
+                gl.shaderSource(shader, source);
+                gl.compileShader(shader);
+                return shader;
+            }
+
+            var vshader_transform = createShader(gl, vertexShaderSourceTransform, gl.VERTEX_SHADER);
```
Comment (pjcozzi) [inline samples/transform_feedback_interleaved.html]: Recall, we are using `camelCase` for variable names.

File: samples/transform_feedback_interleaved.html [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        {
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Init Program
+        var PROGRAM_TRANSFORM = 0;
+        var PROGRAM_FEEDBACK = 1;
+
+        var programs = (function(gl, vertexShaderSourceTransform, fragmentShaderSourceTransform, vertexShaderSourceFeedback, fragmentShaderSourceFeedback) {
```
Comment (pjcozzi) [inline samples/transform_feedback_interleaved.html]: Why is this so complicated?

I get that we want to highlight the use of `transformFeedbackVaryings` in this sample so why not use the `createProgram` helper we already have for the program that isn't using transform feedback, and then cut this code in half and remove the anonymous function and make the GL calls in the main function?

Comment (shrekshao) [inline samples/transform_feedback_interleaved.html]: I think it's because `transformFeedbackVaryings` should be called before `linkProgram`

**Comment by pjcozzi** [comment]
Looks great!  Really just that comment to keep it simple.

File: samples/transform_feedback_interleaved.html (hunk 2 above)
Comment (pjcozzi) [inline samples/transform_feedback_interleaved.html]: Yes, but it is only needed for `program_transform`.  Use our helper function to create the other program.

Comment (shrekshao) [inline samples/transform_feedback_interleaved.html]: Oh yes you are right. Now I get it.

**Comment by pjcozzi** [comment]
Looks great!

**Pull request merged**